File: src/IsbInterop/Accessory/IForEach.vb
﻿Namespace Accessory

  ''' <summary>
  ''' Однонаправленный список.
  ''' </summary>
  Public Interface IForEach(Of Out T As IBaseIsbObject)
    Inherits IBaseIsbObject

    ''' <summary>
    ''' Количество элементов в списке.
    ''' </summary>
    ReadOnly Property Count As Integer

    ''' <summary>
    ''' Признак конца набора данных.
    ''' </summary>
    ReadOnly Property EOF As Boolean

    ''' <summary>
    ''' Получает значение текущего элемента.
    ''' </summary>
    ''' <returns>Значение элемента.</returns>
    Function GetValue() As T

    ''' <summary>
    ''' Перейти к следующему объекту набора данных.
    ''' </summary>
    Sub [Next]()

    ''' <summary>
    ''' Установить указатель текущего элемента списка на первый элемент списка.
    ''' </summary>
    Sub Reset()
  End Interface
End Namespace

File: src/IsbInterop/Collections/IForEach.cs
namespace IsbInterop.Collections
{
  /// <summary>
  /// Однонаправленный список.
  /// </summary>
  public interface IForEach<out T> : IBaseIsbObject where T : IBaseIsbObject
  {
    /// <summary>
    /// Количество элементов в списке.
    /// </summary>
    int Count { get; }

    /// <summary>
    /// Признак конца набора данных.
    /// </summary>
    bool EOF { get; }

    /// <summary>
    /// Получить значение текущего элемента списка.
    /// </summary>
    /// <returns>Значение текущего элемента списка.</returns>
    T GetValue();

    /// <summary>
    /// Перейти к следующему объекту набора данных.
    /// </summary>
    void Next();

    /// <summary>
    /// Установить указатель текущего элемента списка на первый элемент списка.
    /// </summary>
    void Reset();
  }
}
